### PR TITLE
samples: net: zoap: Increase IPv6 multicast addresses

### PR DESCRIPTION
**Pull request by jukkar**
The default IPv6 multicast address count was too small,
increasing addresses to 5.

Jira: ZEP-2560

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

### CODE REVIEW

File: samples/net/zoap_server/prj.conf [hunk 1]
```diff
@@ -9,3 +9,4 @@ CONFIG_TEST_RANDOM_GENERATOR=y
 CONFIG_ZOAP=y
 CONFIG_NET_APP_SETTINGS=y
 CONFIG_NET_APP_MY_IPV6_ADDR="2001:db8::1"
+CONFIG_NET_IF_MCAST_IPV6_ADDR_COUNT=5
```
Comment (nashif) [inline samples/net/zoap_server/prj.conf]: can we set the default to 5 or any safe value and remove the need for guessing the number in many of the same we have?
Right now there are 38 places where this config has different values:

gg NET_IF_MCAST_IPV6_ADDR_COUNT samples/ tests/ | wc -l
38

**Review by nashif — CHANGES_REQUESTED** [review]

File: samples/net/zoap_server/prj.conf (hunk 1 above)
Comment (jukkar) [inline samples/net/zoap_server/prj.conf]: Sure, why not.

Comment (jukkar) [inline samples/net/zoap_server/prj.conf]: I took a better look at this and found out that:
* Each multicast address takes 20 bytes ram, so we increase the ram consumption by 40 bytes if we change the default from 3 to 5. This is a small number but these things seem to add up.
* There is no "safe" value here. The current default 3 is usually quite ok for normal cases, but for some protocols like CoAP that use multicast a lot, that might be too small.

So after second though I suggest we leave the current default to 3 and just increase the number in samples where it makes sense.

Comment (pfalcon) [inline samples/net/zoap_server/prj.conf]: I'm with @jukkar here. 3 should be a "normal" value for "pretty good" IPv6 coverage (definitely enough for 6lowpan devices). Setting the default as 5 seems like waste of memory. Perhaps, we could try "default 5 if NET_COAP", though I yet to confirm that in such case, user still can override such value.

Overall, we have other case where "not good enough" defaults are used, we'd better fix those first before trying things risky on RAM usage in 90% of cases.

Comment (pfalcon) [inline samples/net/zoap_server/prj.conf]: Oh, and btw, we can follow "static assert" approach (#if/#error in this case), that should do just what we need.

**Review by pfalcon — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Pull request merged**